csharp/Samples/Microsoft.Spark.CSharp/DStreamSamples.cs

```diff
@@ -83,6 +88,9 @@ internal static void DStreamTextFileSamples()
                 System.Threading.Thread.Sleep(1000);
             }
 
+            // wait ForeachRDD to complete to let ssc.Stop() gracefully
+            System.Threading.Thread.Sleep(2000);
```

Comment:
actually not, streaming has a separate callback channel not in sync with normal backend channel which might cause conflicts when you shutdown streaming application, not sure it worth the effort at this point (also discussed with Daniel offline), let's ask the question to pyspark guys how they handle it.